### PR TITLE
Feature/release automation

### PR DESCRIPTION
**Pull request by dglaeser**
@timokoch, this is an idea for an automated release process. When a new tag is pushed, this

- checks that the latest tag matches the versions specified in the .cff/cmake files
- generates a release description from the changelog
- builds the doxygen documentation
- creates a release using the description, attaching the built html documentation as asset

### CODE REVIEW

**Comment by dglaeser** [comment]
note: is based on #224

**Comment by timokoch** [comment]
How about turning the check_version script into a set_version script. Then it can be used both to set the version and check the version in the CI via set_version.py && git diff?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# SPDX-FileCopyrightText: 2023 Dennis Gläser <dennis.glaeser@iws.uni-stuttgart.de>
+# SPDX-License-Identifier: MIT
+
+name: releases
+
+on:
```
Comment (timokoch) [inline .github/workflows/release.yml:6]: what's the desired workflow? I update all files with versions, merge to main, and then whenever (possibly much later) I push the tag, the automated release happens?

Comment (timokoch) [reply .github/workflows/release.yml:6]: Do we need to restrict the pipeline to main?

Comment (dglaeser) [reply .github/workflows/release.yml:6]: it's only triggered for tags that start with "v" anyway, so no branches at all. Should be fine I guess?

Comment (dglaeser) [reply .github/workflows/release.yml:6]: I could add a paragraph on this to the main readme in the development section

Comment (dglaeser) [reply .github/workflows/release.yml:6]: I added a paragraph with the envisioned workflow

Comment (dglaeser) [reply .github/workflows/release.yml:6]: I made some modifications to the script & adjusted the readme. From my side this is fine.

**Pull request merged**